rest-service/manager_rest/plugin_update/manager.py

```diff
@@ -0,0 +1,265 @@
+                                                    STATES.EXECUTING_WORKFLOW))
+
+    def _validate_execution_status(self, plugin_update):
+        execution = self.sm.get(models.Execution, plugin_update.execution_id)
+        if execution.status in ExecutionState.END_STATES \
+                and execution.status != ExecutionState.TERMINATED:
+            plugin_update.state = STATES.FAILED
+            self.sm.update(plugin_update)
+            exec_stat_txt = 'failed' if \
+                execution.status == ExecutionState.FAILED else 'cancelled'
+            raise PluginsUpdateError(
+                'The execution of plugins update {0} {1}.'.format(
+                    plugin_update.id,
+                    exec_stat_txt))
+
+    def _get_deployments_to_update(self, blueprint_id):
+        return self.sm.list(models.Deployment,
+                            filters={'blueprint_id': blueprint_id},
+                            sort={'id': 'asc'}).items
+
+    def get_plugins_update(self, plugins_update_id):
+        return self.sm.get(models.PluginsUpdate, plugins_update_id)
+
+    def list_plugins_updates(self,
+                             include=None,
+                             filters=None,
+                             pagination=None,
+                             sort=None,
+                             substr_filters=None):
+        return self.sm.list(models.PluginsUpdate,
+                            include=include,
+                            filters=filters,
+                            pagination=pagination,
+                            substr_filters=substr_filters,
+                            sort=sort)
+
+    def get_reevaluated_plan(self, blueprint):
+
+        blueprint_dir = os.path.join(
+            config.instance.file_server_root,
+            FILE_SERVER_BLUEPRINTS_FOLDER,
+            blueprint.tenant.name,
+            blueprint.id)
+        temp_plan = self.rm.parse_plan(blueprint_dir,
+                                       blueprint.main_file_name,
+                                       config.instance.file_server_root)
+        plan = blueprint.plan
+        if not _did_plugins_to_install_change(temp_plan, plan):
+            raise PluginsUpdateError("No new plugins changes have been "
+                                     "detected when re-parsed blueprint {0}, "
+                                     "aborting update.".format(blueprint.id))
+        return plan
+
+
+def _did_plugins_to_install_change(temp_plan, plan):
+    temp_plugins = temp_plan[DEPLOYMENT_PLUGINS_TO_INSTALL] + \
+                   temp_plan[WORKFLOW_PLUGINS_TO_INSTALL]
+    plugins = plan[DEPLOYMENT_PLUGINS_TO_INSTALL] + plan[
+        WORKFLOW_PLUGINS_TO_INSTALL]
+    temp_plugins_dict = {p[PLUGIN_NAME_KEY]: p for p in temp_plugins}
+    plugins_dict = {p[PLUGIN_NAME_KEY]: p for p in plugins}
+    for p_name in temp_plugins_dict:
+        if temp_plugins_dict[p_name] != plugins_dict.get(p_name, None):
```

Comment:
do this in var loop assignment

Reply:
Don't know why I constructed the temp dict. I removed the redundant dict. I don't think it can be shortened further.